Frontend/src/components/Header.tsx

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable no-console */
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { useState } from 'react';
+import axios from 'axios';
+import { IHeaderProps } from '../interfaces/IHeaderProps';
+
+type FormValues = {
+  newTaskTitle: string;
+};
+
+export default function Header({ tasks, setTasks }: IHeaderProps) {
+  const [newTaskInputVal, setNewTaskInputVal] = useState('');
```

Comment:
Avoid redundancies in naming. The `Val` part is unnecessary, all data stored in the state will be _some_ value, possibly `Input` too. `newTask` and `setNewTask` should work just fine.

Reply:
Done. I was trying to specify that it came from the input, but that does seem a little silly now that you mention it. Thanks for the feedback.